### PR TITLE
feat: Stream data from db, build snapshot incrementally

### PR DESCRIPTION
**Pull request by segfault-magnet**
The snapshot command will now iterate over the chain state and write it in batches to parquet files or JSON, depending on the CLI parameter passed.

The state and balances are still present in `ContractConfig`. They will be removed when we finish the snapshot reading part.

### CODE REVIEW

File: bin/fuel-core/src/cli/snapshot.rs [hunk 1]
```diff
@@ -32,8 +54,11 @@ pub enum SubCommands {
         #[clap(name = "CHAIN_CONFIG", long = "chain", default_value = "local_testnet")]
         chain_config: String,
         /// Specify a path to an output directory for the chain config files.
-        #[clap(name = "OUTPUT_DIR", long = "output directory")]
+        #[clap(name = "OUTPUT_DIR", long = "output_directory")]
```
Comment (Voxelot) [inline bin/fuel-core/src/cli/snapshot.rs:57]:
```suggestion
        #[clap(name = "OUTPUT_DIR", long = "output-directory")]
```

File: bin/fuel-core/src/cli/snapshot.rs [hunk 2]
```diff
@@ -32,8 +54,11 @@ pub enum SubCommands {
...
         output_dir: PathBuf,
+        /// State encoding format
+        #[clap(name = "STATE_ENCODING_FORMAT", long = "state_encoding_format")]
```
Comment (Voxelot) [inline bin/fuel-core/src/cli/snapshot.rs:60]:
```suggestion
        #[clap(name = "STATE_ENCODING_FORMAT", long = "state-encoding-format")]
```

**Review by MitchTurner — COMMENTED** [review]
This is looking good. Is the intention to merge this into master after the parent branch gets merged?

**Comment by segfault-magnet** [comment]
> Is the intention to merge this into master after the parent branch gets merged?

Merging anything into `master` is still a ways off. Our current approach is to have everything land into `feature/regenesis`.

Currently, the PRs look something like this:
feature/regenesis <- PR ( #1459 ) <- PR ( #1474 ) <- PR ($1516) <- PR (#1519)

**Pull request merged**